py-polars/polars/internals/series/series.py

```diff
@@ -897,15 +902,29 @@ def to_frame(self) -> pli.DataFrame:
         │ 1   │
         ├╌╌╌╌╌┤
         │ 2   │
+        └─────┘
+
+        >>> df = s.to_frame("xyz")
+        >>> df
+        shape: (2, 1)
+        ┌─────┐
+        │ xyz │
+        │ --- │
+        │ i64 │
+        ╞═════╡
+        │ 1   │
         ├╌╌╌╌╌┤
-        │ 3   │
+        │ 2   │
         └─────┘
 
-        >>> type(df)
-        <class 'polars.internals.dataframe.frame.DataFrame'>
+        >>> isinstance(df, pl.DataFrame)
+        True
 
         """
-        return pli.wrap_df(PyDataFrame([self._s]))
+        df = pli.wrap_df(PyDataFrame([self._s]))
+        if name:
```

Comment:
We should avoid trouble with empty strings here:

```suggestion
        if name is not None:
```

Reply:
The empty string as a column name is so odd - but you're quite right! Fixed... ;)
(Plus added an explicit test for this case).